### PR TITLE
Frama-C: new beta release (27.0~beta-Cobalt)

### PR DESCRIPTION
**Pull request by AllanBlanchard**

### CODE REVIEW

File: packages/frama-c/frama-c.27.0~beta/opam [hunk 1]
```diff
@@ -0,0 +1,175 @@
+opam-version: "2.0"
```
Comment (AllanBlanchard) [inline packages/frama-c/frama-c.27.0~beta/opam:1]:
```suggestion
opam-version: "2.0"
available: opam-version >= "2.1.0"
flags: avoid-version
```

**Comment by mseri** [comment]
I don't know, but if you have docker you can reproduce it locally (and modify the dockerfile to have a shell in the docker env) by copying the beggining of the logs. For example:
```
cd $(mktemp -d)
git clone --recursive "https://github.com/ocaml/opam-repository.git" && cd "opam-repository" && git fetch origin "refs/pull/23797/head" && git reset --hard 1d1924ba
git fetch origin master
git merge --no-edit 97665ffaf0d03abf4361ba7ffb29ec6d7840b0e6
cat > ../Dockerfile <<'END-OF-DOCKERFILE'
FROM ocaml/opam:debian-10-ocaml-4.14@sha256:1cad619eaca9ad2aab98ea766efbccd33f7506d88ede0f7abbd8cdd37d41679f
USER 1000:1000
WORKDIR /home/opam
RUN sudo ln -f /usr/bin/opam-dev /usr/bin/opam
RUN opam init --reinit -ni
ENV OPAMDOWNLOADJOBS="1"
ENV OPAMERRLOGLEN="0"
ENV OPAMSOLVERTIMEOUT="500"
ENV OPAMPRECISETRACKING="1"
RUN rm -rf opam-repository/
COPY --chown=1000:1000 . opam-repository/
RUN opam repository set-url --strict default opam-repository/
RUN opam update --depexts || true
ENV OPAMCRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
ENV OPAMFIXUPCRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
ENV OPAMUPGRADECRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
RUN opam pin add -k version -yn frama-c.27.0~beta 27.0~beta
RUN opam reinstall frama-c.27.0~beta; \
    res=$?; \
    test "$res" != 31 && exit "$res"; \
    export OPAMCLI=2.0; \
    build_dir=$(opam var prefix)/.opam-switch/build; \
    failed=$(ls "$build_dir"); \
    partial_fails=""; \
    for pkg in $failed; do \
    if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"debian-10\""; then \
    echo "A package failed and has been disabled for CI using the 'x-ci-accept-failures' field."; \
    fi; \
    test "$pkg" != 'frama-c.27.0~beta' && partial_fails="$partial_fails $pkg"; \
    done; \
    test "${partial_fails}" != "" && echo "opam-repo-ci detected dependencies failing: ${partial_fails}"; \
    exit 1
ENV OPAMCRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
ENV OPAMFIXUPCRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
ENV OPAMUPGRADECRITERIA="-removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed"
RUN (opam reinstall --with-test frama-c.27.0~beta) || true
RUN opam reinstall --with-test --verbose frama-c.27.0~beta; \
    res=$?; \
    test "$res" != 31 && exit "$res"; \
    export OPAMCLI=2.0; \
    build_dir=$(opam var prefix)/.opam-switch/build; \
    failed=$(ls "$build_dir"); \
    partial_fails=""; \
    for pkg in $failed; do \
    if opam show -f x-ci-accept-failures: "$pkg" | grep -qF "\"debian-10\""; then \
    echo "A package failed and has been disabled for CI using the 'x-ci-accept-failures' field."; \
    fi; \
    test "$pkg" != 'frama-c.27.0~beta' && partial_fails="$partial_fails $pkg"; \
    done; \
    test "${partial_fails}" != "" && echo "opam-repo-ci detected dependencies failing: ${partial_fails}"; \
    exit 1

END-OF-DOCKERFILE
docker build -f ../Dockerfile .
```

**Comment by maroneze** [comment]
Thanks, we're testing it using ocaml/opam's Docker images, and indeed with GCC 7.5 (present in Ubuntu 18.04 and opensuse 15.4), the warning that we had seen in Clang is also present. With GCC 9 and newer, we don't see it.

We'll see if we can find a solution to avoid these warnings.

**Comment by AllanBlanchard** [comment]
Anyway, I think that we are OK for this beta release :)

**Pull request merged**